src/sentry/tasks/assemble.py

```diff
@@ -799,55 +599,55 @@ def assemble_artifacts(
     # These params have been added for supporting artifact bundles assembling.
     project_ids=None,
     dist=None,
-    upload_as_artifact_bundle=False,
     is_release_bundle_migration=False,
     **kwargs,
-):
+) -> None:
     """
     Creates a release file or artifact bundle from an uploaded bundle given the checksums of its chunks.
     """
     # We want to evaluate the type of assemble task given the input parameters.
-    assemble_task = (
-        AssembleTask.ARTIFACT_BUNDLE if upload_as_artifact_bundle else AssembleTask.RELEASE_BUNDLE
-    )
+    assemble_task = AssembleTask.ARTIFACT_BUNDLE
 
     try:
         organization = Organization.objects.get_from_cache(pk=org_id)
         bind_organization_context(organization)
 
         set_assemble_status(assemble_task, org_id, checksum, ChunkFileState.ASSEMBLING)
 
-        archive_name = "bundle-artifacts" if upload_as_artifact_bundle else "release-artifacts"
-        archive_filename = f"{archive_name}-{uuid.uuid4().hex}.zip"
-        file_type = "artifact.bundle" if upload_as_artifact_bundle else "release.bundle"
-
         # Assemble the chunks into a temporary file
         assemble_result = assemble_file(
             task=assemble_task,
             org_or_project=organization,
-            name=archive_filename,
+            name=f"bundle-artifacts-{uuid.uuid4().hex}.zip",
             checksum=checksum,
             chunks=chunks,
-            file_type=file_type,
+            file_type="artifact.bundle",
         )
 
         # If not file has been created this means that the file failed to assemble because of bad input data.
         # In this case, assemble_file has set the assemble status already.
         if assemble_result is None:
             return
 
+        if not project_ids:
+            raise AssembleArtifactsError(
+                "uploading an artifact bundle without a project is prohibited"
+            )
+
         # We first want to prepare the post assembler which will take care of validating the archive.
-        with prepare_post_assembler(
+        post_assembler = ArtifactBundlePostAssembler(
             assemble_result=assemble_result,
             organization=organization,
             release=version,
             dist=dist,
             project_ids=project_ids,
-            upload_as_artifact_bundle=upload_as_artifact_bundle,
             is_release_bundle_migration=is_release_bundle_migration,
-        ) as post_assembler:
+        )
+
+        with post_assembler:
```

Comment:
What does `with` do if you created a binding for `post_assembler` on a previous line?

Reply:
The context manager will still make sure that the invalid file is being deleted when any kind of error happens.